### PR TITLE
MLE-3943: Updating the pipeline to use marklogic docker

### PR DESCRIPTION
**Pull request by SameeraPriyathamTadikonda**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -53,7 +51,13 @@ pipeline{
       }
       agent {label 'devExpLinuxPool'}
       steps{
-        runtests('Latest','11','JAVA11_HOME_DIR')
+        sh label:'mlsetup', script: '''#!/bin/bash
+            sudo /usr/local/sbin/mladmin remove
```
Comment (rjrudin) [inline Jenkinsfile:55]: What do these two lines do? I'm thinking that since they're repeated below, it'd be good to put them into a function that has a comment on it explaining the purpose of these lines.

Comment (SameeraPriyathamTadikonda) [reply Jenkinsfile:57]: Added a print statement.

Line 54 - Opens a shell terminal and 55 removes any existing marklogic.

File: Jenkinsfile [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
-  copyRPM mlVersionType,mlVersion
-  setUpML '$WORKSPACE/xdmp/src/Mark*.rpm'
+def runtests(String javaVersion){
   sh label:'test', script: '''#!/bin/bash
     export JAVA_HOME=$'''+javaVersion+'''
     export GRADLE_USER_HOME=$WORKSPACE/$GRADLE_DIR
     export PATH=$GRADLE_USER_HOME:$JAVA_HOME/bin:$PATH
     cd marklogic-spark-connector
     echo "mlPassword=admin" > gradle-local.properties
+    sleep 30s
```
Comment (rjrudin) [inline Jenkinsfile:10]: Why is this here? If it's needed, add a comment before it to explain why it's sleeping and why 30s is the chosen value.

File: Jenkinsfile [hunk 3]
```diff
@@ -53,11 +52,29 @@ pipeline{
       }
       agent {label 'devExpLinuxPool'}
       steps{
-        runtests('Latest','11','JAVA11_HOME_DIR')
+        sh label:'mlsetup', script: '''#!/bin/bash
+            echo "Removing any running marklogic server and clean up marklogic data directory"
```
Comment (rjrudin) [inline Jenkinsfile:56]: Small nit - always do "MarkLogic" in a comment/printout, not "marklogic".

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 3 above)
Comment (SameeraPriyathamTadikonda) [reply Jenkinsfile]: Done

File: Jenkinsfile (hunk 2 above)
Comment (SameeraPriyathamTadikonda) [reply Jenkinsfile:11]: Added a comment.

**Review by rjrudin — APPROVED** [review]
